pkg/deploy/server/default_values_reconciler.go

```diff
@@ -0,0 +1,274 @@
+//
+// Copyright (c) 2019-2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package server
+
+import (
+	"strconv"
+
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+	"github.com/eclipse-che/che-operator/pkg/util"
+	"github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type DefaultValuesReconciler struct {
+	deploy.Reconcilable
+}
+
+func NewDefaultValuesReconciler() *DefaultValuesReconciler {
+	return &DefaultValuesReconciler{}
+}
+
+func (p *DefaultValuesReconciler) Reconcile(ctx *deploy.DeployContext) (reconcile.Result, bool, error) {
+	cheFlavor := deploy.DefaultCheFlavor(ctx.CheCluster)
+	cheNamespace := ctx.CheCluster.Namespace
+	if len(ctx.CheCluster.Spec.Server.CheFlavor) < 1 {
```

Comment:
Up to you, but for me looks better len(ctx.CheCluster.Spec.Server.CheFlavor) == 0